install_config/upgrading/migrating_etcd.adoc

```diff
@@ -0,0 +1,274 @@
+outage of the API, web console, and controllers.
+
+This migration process performs the following steps:
+
+- Stop master API and controller services
+- Perform an etcd backup on all etcd members
+- Perform a migration on the first etcd host
+- Remove etcd data from remaining etcd hosts
+- Perform an etcd scaleup operation adding additional etcd hosts one by one
+- Re-introduce TTL information on specific keys
+- Reconfigure masters for etcd v3 storage
+- Start master API and controller services
+
+[[etcd-data-migration-before-you-begin]]
+== Before You Begin
+
+You can only begin the etcd data migration process after upgrading to
+{product-title} 3.6, as previous versions are not compatible with etcd v3
+storage. Additionally, the upgrade to {product-title} 3.6 reconfigures cluster
+DNS services to run on every node, rather than on the masters, which ensures
+that, even when master services are taken down, existing pods continue to
+function as expected.
+
+The migration process is currently only supported on clusters that have etcd
+hosts specifically defined in their inventory file, therefore it cannot be used
+for clusters which utilize the embedded etcd which runs as part of the master
+process. Support for migrating embedded etcd installations will be added in a
+future release.
+
+[[etcd-data-migration-automated]]
+== Running the Automated Migration Playbook
+
+A migration playbook is provided to automate all aspects of the process; this is the preferred method for performing the migration. You must have access
+to your existing inventory file with both masters and etcd hosts defined in their separate groups. 
+
+. Ensure you have the latest version of the *openshift-ansible* packages
+installed:
++
+----
+# yum upgrade openshift-ansible\*
+----
+
+. Run the *_migrate.yml_* playbook using your inventory file:
++
+----
+# ansible-playbook [-i /path/to/inventory] \
+ifdef::openshift-enterprise[]
+    /usr/share/ansible/openshift-ansible/playbooks/byo/openshift-etcd/migrate.yml
+endif::[]
+ifdef::openshift-origin[]
+    ~/openshift-ansible/playbooks/byo/openshift-etcd/migrate.yml
+endif::[]
+----
+
+[[etcd-data-migration-manual]]
+== Running the Migration Manually
+
+The following procedure describes the steps required to successfully migrate the
+cluster (implemented as part of the Ansible etcd migration playbook).
+
+. +++<b>Stop masters and wait for etcd convergence:</b>+++
+
+.. Stop all master services:
++
+----
+# systemctl stop atomic-openshift-master \
+    atomic-openshift-master-api \
+    atomic-openshift-master-controllers
+----
+
+.. Before the migration can proceed, the etcd cluster must be healthy
+and raft indices of all etcd members must differ by one unit at most.
+At the same time, all etcd members and master daemons must be stopped.
++
+To check the etcd cluster is healthy you can run:
++
+----
+# etcdctl <certificate_details> <endpoint> cluster-health
```

Comment:
Do users know what certificate_details are?

Reply:
@sdodson Meant to ask about this. Maybe link to https://docs.openshift.com/container-platform/3.6/admin_guide/backup_restore.html#restoring-external-etcd as an example? It shows the cert flags in there.

Reply:
The `<certificate_details>` is a common pattern for certs all over the place. Though, saying that the link will surely help. I would rather use https://docs.openshift.com/container-platform/3.6/admin_guide/backup_restore.html#adding-addtl-etcd-members to point to the cert flags with something like "example of how to set the certificate flags can be seen at <link>".